OIDC Login

dasGoogle opened the pull request:
Fixes #59 

Description (what might a PO or reviewer want to know?)
- Log in using the OIDC Login button on the login page
- You can keep your current account if the email address matches

## PR checklist

- [ ] dev-branch has been merged into local branch to resolve conflicts
- [ ] tests and linter have passed AFTER local merge
- [ ] localization is supported [(Guide)](https://github.com/hpi-swt2/bookkeeper-portal-blue/wiki/Internationalization-guide)
- [ ] another dev reviewed and approved
- [ ] if feature-Branch: Teams PO has approved (show via e.g. screenshots/screencapture/live demo)

dasGoogle commented:
[Peek 2023-01-05 16-21.webm](https://user-images.githubusercontent.com/16378455/210815510-ef5857f1-fe62-484a-8951-2da114e62926.webm)

@benn02

dasGoogle commented:
@rgwohlbold When deploying OIDC, we need to add OIDC client credentials in the corresponding ENV variables. See [here](https://github.com/hpi-swt2/bookkeeper-portal-blue/blob/b4c4c46224cc321f433754f3e2c2427b06054eae/config/initializers/devise.rb#L275). Can you take care of this?

rgwohlbold commented:
> @rgwohlbold When deploying OIDC, we need to add OIDC client credentials in the corresponding ENV variables. See [here](https://github.com/hpi-swt2/bookkeeper-portal-blue/blob/b4c4c46224cc321f433754f3e2c2427b06054eae/config/initializers/devise.rb#L275). Can you take care of this?

I created two OIDC clients and added you as an owner @dasGoogle. I added the config variables to both deployments, we should check after the merge that it is configured correctly.

dasGoogle commented:
> > @rgwohlbold When deploying OIDC, we need to add OIDC client credentials in the corresponding ENV variables. See [here](https://github.com/hpi-swt2/bookkeeper-portal-blue/blob/b4c4c46224cc321f433754f3e2c2427b06054eae/config/initializers/devise.rb#L275). Can you take care of this?
> 
> I created two OIDC clients and added you as an owner @dasGoogle. I added the config variables to both deployments, we should check after the merge that it is configured correctly.

Thanks!

lukasrad02 commented on app/views/devise/shared/_links.html.erb:
```diff
@@ -1,3 +1,14 @@
+    <%= button_to(omniauth_authorize_path(resource_name, provider),
+                  class: 'btn btn-primary', id: "#{provider}-signin" ,
+                  method: :post,
+                  data: {disable_with: "<i class='fa fa-spinner fa-spin'></i> Logging in...", turbo: "false"}) do %>
```
If I understand this correctly, the "logging in" text will be shown on the "Sign in with OIDC" button after clicking it, but I've been never able to see it, even with throttled requests to the OIDC server.

But if it should be visible in some cases, it would be nice if it was localized.

dasGoogle replied:
I simply removed this text now, I think the icon suffices. I honestly also don't know when this should be shown.

lukasrad02 reviewed (commented):
Besides the localization of the button, it looks pretty good to me. Nice work!

dasGoogle commented:
> > @rgwohlbold When deploying OIDC, we need to add OIDC client credentials in the corresponding ENV variables. See [here](https://github.com/hpi-swt2/bookkeeper-portal-blue/blob/b4c4c46224cc321f433754f3e2c2427b06054eae/config/initializers/devise.rb#L275). Can you take care of this?
> 
> I created two OIDC clients and added you as an owner @dasGoogle. I added the config variables to both deployments, we should check after the merge that it is configured correctly.

It seems that the base URL you configured in `APP_BASE_URL` is http rather than https which makes the authentication fail. Could you maybe fix this in the deployments, @rgwohlbold?

rgwohlbold commented:
> should
Sorry, should bei fixed now
